### PR TITLE
Release notes for 1.3.0

### PR DESCRIPTION
**Pull request by straight-shoota**
I'm starting this PR early to improve and untangle the release process. We already know most of the things that are going to be released in 1.3.0. The major changes should already be in master. So we can start writing about them.
There is a similar PR for the changelog: https://github.com/crystal-lang/crystal/pull/11599

The following is a list of the most important changes that I think should be mentioned in the release notes. If you have additional suggestions, please comment. The list is open for extensions.

* [x] Interpreter (@asterite, @straight-shoota) (#229)
* [x] Windows Support (@HertzDevil) (https://github.com/crystal-lang/crystal-website/pull/230)
* [x] Unicode (@straight-shoota) (https://github.com/crystal-lang/crystal-website/pull/228)
* [x] Syscalls (@lbguilherme) (https://github.com/crystal-lang/crystal-website/pull/231)
* [x] Multi-Assign (@HertzDevil) (#227)
* [x] Number Autocast (@asterite, @beta-ziliani) (#225)
* [x] Improved Numerical Support / Lexing Number Literals (@BlobCodes, @HertzDevil) (https://github.com/crystal-lang/crystal-website/pull/232)

I tagged some people who I believe did major contributions for the respective changes in this release and hope they might also contribute a brief description for the release notes. Don't feel obliged to do that. I just think whoever implemented something is also best qualified to tell about it. Anyone else can step in as well.
Descriptions should summarize the change and for whom it's relevant. A code example is often helpful. You can draw inspiration from [previous release notes](https://crystal-lang.org/blog/#release_notes). Please submit via pull request against this branch (https://github.com/crystal-lang/crystal-website/tree/release/1.3.0).

### CODE REVIEW

**Comment by beta-ziliani** [comment]
Perhaps a specific item for the `**(breaking-change)**`s not listed in the other items?

**Review by beta-ziliani — APPROVED** [review]

File: _releases/2022-01-06-1.3.0-released.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+## Number Autocast
+
+Primitive numeric values are now _autocasted_ to fit into larger types ([#11431](https://github.com/crystal-lang/crystal/pull/11431), [#11529](https://github.com/crystal-lang/crystal/pull/11529)). For instance, it is possible to call a function expecting an `Int64` with an `Int32` (note that before 1.3.0 only number _literals_ were autocasted):
+
+<div class="code_section">{% highlight crystal %}
+def foo(x : Int64)
+  x
+end
+
+foo 1_32 # Works in 1.2.2 and 1.3.0
```
Comment (BlobCodes) [inline _releases/2022-01-06-1.3.0-released.md:154]: A little late, but shouldn't this be `1_i32`?

Comment (beta-ziliani) [reply _releases/2022-01-06-1.3.0-released.md:154]: 🤦

Comment (asterite) [reply _releases/2022-01-06-1.3.0-released.md:154]: Technically `1_32` is still an Int32, so it's fine 😅

**Pull request merged**